packages/debug/src/browser/view/debug-configuration-select.tsx

```diff
@@ -0,0 +1,223 @@
+    private quickInputService: QuickInputService;
+
+    constructor(props: DebugConfigurationSelectProps) {
+        super(props);
+        this.manager = props.manager;
+        this.quickInputService = props.quickInputService;
+        this.state = {
+            providerTypes: [],
+        };
+        this.manager.onDidChangeConfigurationProviders(() => {
+            this.refreshDebugConfigurations();
+        });
+    }
+
+    componentDidMount(): void {
+        this.refreshDebugConfigurations();
+    }
+
+    render(): React.ReactNode {
+        return <select
+            className='theia-select debug-configuration'
+            value={this.currentValue}
+            onChange={this.setCurrentConfiguration}
+            onFocus={this.refreshDebugConfigurations}
+            onBlur={this.refreshDebugConfigurations}
+        >
+            {this.renderOptions()}
+        </select>;
+    }
+
+    protected get currentValue(): string {
+        const { current } = this.manager;
+        return current ? InternalDebugSessionOptions.toValue(current) : DebugConfigurationSelect.NO_CONFIGURATION;
+    }
+
+    protected readonly setCurrentConfiguration = (event: React.ChangeEvent<HTMLSelectElement>) => {
+        const value = event.currentTarget.value;
+        if (value === DebugConfigurationSelect.ADD_CONFIGURATION) {
+            this.manager.addConfiguration();
+        } else if (value.startsWith(DebugConfigurationSelect.PICK)) {
+            const providerType = this.parsePickValue(value);
+            this.selectDynamicConfigFromQuickPick(providerType);
+        } else {
+            const [name, type, request, workspaceFolderUri, providerType] = InternalDebugSessionOptions.parseValue(value);
+            this.manager.current = this.manager.find(
+                { name, type, request },
+                workspaceFolderUri,
+                providerType === 'undefined' ? undefined : providerType
+            );
+        }
+    };
+
+    protected toPickValue(providerType: string): string {
+        return DebugConfigurationSelect.PICK + providerType;
+    }
+
+    protected parsePickValue(value: string): string {
+        return value.slice(DebugConfigurationSelect.PICK.length);
+    }
+
+    protected async resolveDynamicConfigurationPicks(providerType: string): Promise<DynamicPickItem[]> {
+        const configurationsOfProviderType =
+            (await this.manager.provideDynamicDebugConfigurations()).find(entry => entry.type === providerType);
+
+        const picks: DynamicPickItem[] = [];
+        if (!configurationsOfProviderType) {
+            return picks;
+        }
+
+        const { configurations } = configurationsOfProviderType;
+        for (const configuration of configurations) {
+            picks.push({
+                label: configuration.name,
+                configurationType: configuration.type,
+                request: configuration.request,
+                providerType
+            });
+        }
```

Comment:
Matter of taste only:
```suggestion
        if (!configurationsOfProviderType) {
            return [];
        }

        return configurationsOfProviderType.configurations.map(configuration => ({
            label: configuration.name,
            configurationType: configuration.type,
            request: configuration.request,
            providerType
        }));
```